### ISSUE TITLE
Hitting enter in a text box adds a space

### ISSUE THREAD

**Issue by cbrewster** [servo/servo#20630]
Test Case:
`<input type="text"/>`

1. Type in textbox
2. Hit enter
3. Space is added to end of text

**Comment by paulrouget** [comment]
If it's a regression, 2 PRs to check: #19895 and #20327

-

**Comment by atouchet** [comment]
I don't think this is a recent regression at the very least. I've been able to reproduce the bug in builds going back to mid-2017. I haven't tested past that.

**Comment by kwonoj** [comment]
I could see some pictures around why this is happening (note: appended char is `\r`, not space) but need to simmer bit more time to present some PR to discuss further. I'll try bit more.

**Comment by yezhizhen** [comment]
This problem no longer occur. Can someone close it?

**Comment by yezhizhen** [comment]
This problem no longer occurs. Can someone close it?
